_rules/summary-non-empty-accessible-name-2t702h.md

```diff
@@ -0,0 +1,195 @@
+---
+id: 2t702h
+name: Summary element has non-empty accessible name
+rule_type: atomic
+description: |
+  This rule checks that each `summary` element has a non-empty accessible name.
+accessibility_requirements:
+  wcag20:4.1.2: # Name, Role, Value (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - Accessibility Tree
+  - DOM Tree
+  - CSS Styling
+acknowledgments:
+  authors:
+    - Wilco Fiers
+---
+
+## Applicability
+
+This rule applies to HTML `summary` elements for which all the following is true:
+
+- the element is [included in the accessibility tree][]; and
+- the element works as the [summary button for its parent `details` element][details summary]; and
+- the element has no [explicit role][], or the [explicit role][] is ignored due to a [presentational role conflict][].
+
+## Expectation
+
+Each target element has an [accessible name][] that is not empty (`""`).
+
+## Assumptions
+
+The rule assumes that all `summary` elements are [user interface components as defined by WCAG 2](https://www.w3.org/TR/WCAG22/#dfn-user-interface-components).
+
+## Accessibility Support
+
+Implementation of [Presentational Roles Conflict Resolution][] varies from one browser or assistive technology to another. Depending on this, some elements can have a [semantic role][] of `button` and fail this rule with some technology but users of other technologies would not experience any accessibility issue.
```

Comment:
Why would the element having a semantic role of button make it fail this rule? Is this copy/paste from another rule?